src/Microsoft.AspNet.Owin/OwinFeatureCollection.cs

```diff
@@ -427,6 +428,20 @@ public bool IsReadOnly
             get { return true; }
         }
 
+        Guid IRequestIdentifierFeature.TraceIdentifier
+        {
+            get
+            {
+                var requestId = Prop<string>(OwinConstants.RequestId);
+                if (requestId != null)
+                {
+                    return new Guid(requestId);
```

Comment:
The OWIN spec say it's an opaque value, we can't assume it will parse as a GUID. TryParse?